### PR TITLE
Ft main deparser task stubs

### PR DESCRIPTION
**Pull request by mpbtruong**
Please review my nonsense

### CODE REVIEW

File: main-application/comms-processing_tasks.c
Comment (tyrelkostyk) [inline main-application/comms-processing_tasks.c]: - I don't think inline comments need the full /* */ syntax; just a single // is fine IMO (we can confirm this later)
- This will block forever until data is available, FYI (comment should reflect that)
- Forgot extra space after portMAX_DELAY

File: main-application/comms-processing_tasks.c
Comment (tyrelkostyk) [inline main-application/comms-processing_tasks.c]: Just saying: in practice these will likely be a custom type, not just a uint16_t. Also, what do you think about being even more explicit and naming these `rawOutgoingPacket`, etc?

File: main-application/comms-processing_tasks.c
Comment (tyrelkostyk) [inline main-application/comms-processing_tasks.c]: You take a Semaphore, but never give it back. Is this intentional? Should event groups be used, or task notifications? Just thinking out load

File: main-application/comms-processing_tasks.c
Comment (tyrelkostyk) [inline main-application/comms-processing_tasks.c]: use relative includes with `<` and `>` instead of quotes

File: main-application/comms-processing_tasks.c
Comment (tyrelkostyk) [inline main-application/comms-processing_tasks.c]: Are you sure you want `&deparsedPacket`? Did you mean `*deparsedPacket` or `deparsedPacket`? What you have would likely be illegal; that's the syntax to reassign a pointer. Are these variables pointers, or just local variables? Also, if deparsedPacket is a pointer, then consider having those stub functions return success or status values (since the packet will be updated directly, since we'll be passing it by reference instead of by value).

File: main-application/comms-processing_tasks.c
Comment (tyrelkostyk) [inline main-application/comms-processing_tasks.c:31]: Rather than "FUNCTIONS", for functions I usually use one of the following (depending what's in the following code)
- PUBLIC API
- PRIVATE FUNCTIONS
- PRIVATE FUNCTION STUBS
- FREERTOS TASKS

Seems like in this case, the last one would be most appropriate.

File: main-application/comms-processing_tasks.c
Comment (tyrelkostyk) [inline main-application/comms-processing_tasks.c]: Are these globals intended to be accessible by other files? If so, then you'll need to add these stubs to the header file, and I think they'll need to be prefixed with the `extern` keyword (extern means "defined elsewhere"). Not sure though; I do know that if a different source file wanted to use these, then they'd need to define them with `extern` keyword.

However, if these are meant to be private and only accessible by this file, then you'll need to prefix them with the `static` keyword to prevent other files from accessing them (either accidentally or on purpose)

File: main-application/comms-processing_tasks.h
Comment (tyrelkostyk) [inline main-application/comms-processing_tasks.h]: I usually call these:
- DEFINITIONS (instead of macros)
- PUBLIC GLOBALS
- PUBLIC API STUBS

but that's just my preference lol. If there are no objections I'll likely add these as recommendations to the readme, for the style guide

File: main-application/comms-processing_tasks.c
Comment (tyrelkostyk) [inline main-application/comms-processing_tasks.c]: See following comments.
I would usually call this PRIVATE GLOBALS since you'd usually put globals you'd only want available to _this_ file here.

File: main-application/comms-processing_tasks.c
Comment (tyrelkostyk) [inline main-application/comms-processing_tasks.c]: Interesting approach to always yielding execution immediately after entering the task, before running. Is there anything else that this task should be blocked by? i.e. any specific event group flags, or states? etc

File: main-application/comms-processing_tasks.c
Comment (tyrelkostyk) [inline main-application/comms-processing_tasks.c]: How do you feel about the name deparse? Personally I hate it even though I think I was the one who coined it haha. How about encodeTask?

**Review by tyrelkostyk — CHANGES_REQUESTED** [review]
Great first draft! After some cleanup it can be merged in. After that, I'd recommend that the next steps are fleshing out the FreeRTOS logic, and tying the tasks into the supporting functions.

File: main-application/comms-processing_tasks.c
Comment (mpbtruong) [reply main-application/comms-processing_tasks.c]: Should this be blocking in the first place? Please let me know if you have any thoughts on other implementations!

File: main-application/comms-processing_tasks.c
Comment (mpbtruong) [reply main-application/comms-processing_tasks.c]: Do we know what the custom type name may be yet?

File: main-application/comms-processing_tasks.c
Comment (mpbtruong) [reply main-application/comms-processing_tasks.c]: Done

File: main-application/comms-processing_tasks.c
Comment (mpbtruong) [reply main-application/comms-processing_tasks.c]: Done

File: main-application/comms-processing_tasks.c
Comment (mpbtruong) [reply main-application/comms-processing_tasks.c]: Thanks, these should be external variables.

Comment (mpbtruong) [reply main-application/comms-processing_tasks.c]: EDIT: Also I believe the extern keyword when then be used in the header?

File: main-application/comms-processing_tasks.h
Comment (mpbtruong) [reply main-application/comms-processing_tasks.h]: Ack

File: main-application/comms-processing_tasks.c
Comment (mpbtruong) [reply main-application/comms-processing_tasks.c]: Ack

File: main-application/comms-processing_tasks.c
Comment (mpbtruong) [reply main-application/comms-processing_tasks.c]: I believe this task should only be running if there are raw packets in the receive queue ( rawPacketReceiveQueue ) that need to be deparsed, otherwise the task should yield immediately. I think the implementation for this is to just make a conditional to check if there is anything to be deparsed. Thoughts?

File: main-application/comms-processing_tasks.c
Comment (tyrelkostyk) [reply main-application/comms-processing_tasks.c]: Not yet, so for now it's fine

File: main-application/comms-processing_tasks.c
Comment (tyrelkostyk) [reply main-application/comms-processing_tasks.c]: I would agree

File: main-application/comms-processing_tasks.c
Comment (tyrelkostyk) [reply main-application/comms-processing_tasks.c]: It should be blocking on this, yeah. All tasks need to be at least partially blocking so that the idle task (FreeRTOS's "cleanup" / background task) has execution time to run

**Review by rarohel — APPROVED** [review]
Looks good, pending the changes that Tyrel requested.

File: main-application/comms-processing_tasks.c
Comment (mpbtruong) [reply main-application/comms-processing_tasks.c]: Added xSemaphoreGive call below.

**Pull request closed without merging**
